lib/internal/util.js

```diff
@@ -850,10 +850,34 @@ const encodingsMap = { __proto__: null };
 for (let i = 0; i < encodings.length; ++i)
   encodingsMap[encodings[i]] = i;
 
+/**
+ * Reassigns the .name property of a function.
+ * Should be used when function can't be initially defined with desired name
+ * or when desired name should include `#`, `[`, `]`, etc.
+ * @param {string | symbol} name
+ * @param {Function} fn
+ * @param {object} [descriptor]
+ * @returns {Function} the same function, renamed
+ */
+function assignFunctionName(name, fn, descriptor = kEmptyObject) {
+  return ObjectDefineProperty(fn, 'name', {
+    __proto__: null,
+    writable: false,
+    enumerable: false,
+    configurable: true,
+    ...ObjectGetOwnPropertyDescriptor(fn, 'name'),
+    ...descriptor,
+    value: typeof name === 'symbol' ?
+      (name.description !== undefined ? `[${name.description}]` : '') :
```

Comment:
```suggestion
      (SymbolPrototypeGetDescription(name) !== undefined ? `[${name.description}]` : '') :
```
Can you store it in a variable so we call the getter just once?